### PR TITLE
Event Settings Screen UI + Basic Functions

### PR DESCRIPTION
**Pull request by ZoroarkZark**
## Problem/Purpose
This pull request provides a user interface for the event settings screen and a duration screen that shows the current implementation of the duration picker. 

## Proposed Solution
Two views have been added, one for the duration screen, and one for the general event settings. 

The event settings screen includes a switch for hiding an event after the start date by default, and two buttons to navigate to different screens where you would either select the default location or edit duration presets.

The duration screen includes the default event durations, displayed in a 2x3 grid format, with a "+" button to add another duration being included at 5 or below presets.

## Results
With this pull request, a working event settings screen alongside the current duration implementation should be included, with demonstrations of how the screens would look currently in Storybook.

## Tickets
TASK_itDah9Qi

### CODE REVIEW

**Comment by ZoroarkZark** [comment]
[Event Setting Screen UI](https://trello.com/c/itDah9Qi/644-event-setting-screen-ui)

File: settings-boundary/EventSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+            {createDurationCards(3, sortedDurations)}
+            {settings.eventPresetDurations.length < 6 ? (
+              <AddDurationCard />
+            ) : undefined}
+            {settings.eventPresetDurations.length < 5 ? (
+              <View style={[styles.container, { opacity: 0 }]} />
+            ) : undefined}
+            {settings.eventPresetDurations.length < 4 ? (
+              <View style={[styles.container, { opacity: 0 }]} />
+            ) : undefined}
```
Comment (ZoroarkZark) [inline settings-boundary/EventSettings.tsx:87]: Be sure to have this be more general (rows and columns / grid component with items)

File: settings-storage/PersistentStore.ts [hunk 2]
```diff
@@ -86,6 +86,7 @@ export class PersistentSettingsStore<Settings extends AnySettings>
     if (this.initialLoadPromise) return
     this.initialLoadPromise = this.persistedSettings()
       .then((settings) => {
+        console.log(settings)
```
Comment (ZoroarkZark) [inline settings-storage/PersistentStore.ts:89]: Remove console.log

File: .idea/workspace.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (seanim0920) [inline .idea/workspace.xml:1]: I presume this file isn't needed

File: settings-boundary/EventSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+      title="Presets"
+      subtitle="These presets will be populated when you create an event."
+    >
+      <SettingsNamedToggleView
+        name={"Hide After Start Date"}
+        description={
+          "The event will no longer be publicly visible to other users once it starts."
+        }
+        isOn={settings.eventPresetShouldHideAfterStartDate}
+        onIsOnChange={(eventPresetShouldHideAfterStartDate) =>
```
Comment (seanim0920) [inline settings-boundary/EventSettings.tsx:228]: unrelated to this pr but the name onIsOnChange is a confusing prop name, I'd maybe call it "enabled" and "onEnabledChange" or something similar

**Review by seanim0920 — APPROVED** [review]

**Review by mhayes853 — APPROVED** [review]

**Pull request merged**